flink-core/src/main/java/org/apache/flink/api/common/typeutils/TypeSerializerSerializationUtil.java

```diff
@@ -102,17 +99,24 @@
 	 *
 	 * @return the deserialized serializer.
 	 */
-	public static <T> TypeSerializer<T> tryReadSerializer(DataInputView in, ClassLoader userCodeClassLoader, boolean useDummyPlaceholder) {
+	public static <T> TypeSerializer<T> tryReadSerializer(
+			DataInputView in,
+			ClassLoader userCodeClassLoader,
+			boolean useDummyPlaceholder) throws IOException {
+
 		final TypeSerializerSerializationUtil.TypeSerializerSerializationProxy<T> proxy =
-			new TypeSerializerSerializationUtil.TypeSerializerSerializationProxy<>(userCodeClassLoader, useDummyPlaceholder);
+			new TypeSerializerSerializationUtil.TypeSerializerSerializationProxy<>(userCodeClassLoader);
 
 		try {
 			proxy.read(in);
 			return proxy.getTypeSerializer();
-		} catch (IOException e) {
-			LOG.warn("Deserialization of serializer errored; replacing with null.", e);
-
-			return null;
+		} catch (UnloadableTypeSerializerException e) {
```

Comment:
I would let this bubble up one more level, remove the flag here and only catch `UnloadableTypeSerializerException ` in the case where this method is called with `true`.